test/garbage_collection_test.js

```diff
@@ -26,7 +26,12 @@ suite('garbage collection tests', function () {
       delayFactor: 15 * 1000,
       randomizationFactor: 0.25
     },
-    log: logger()
+    log: logger(),
+    stats: {
+      timeGen: async function (series, fn) {
+        return await fn;
```

Comment:
Would checking fn.then == function be enough to assert that it's a promise in this case?